src/docfx/docset/FilePath.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Docs.Build
+{
+    internal class FilePath : IEquatable<FilePath>, IComparable<FilePath>
+    {
+        /// <summary>
+        /// The file relative path
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// Gets the value to indicate where is this file from
+        /// </summary>
+        public FileOrigin Origin { get; }
+
+        public FilePath(string path, Docset docset)
```

Comment:
This class will become a very primitive type to be consumed by a lot of places, so consider:
- move this file to `lib`
- remove the dependency on `Docset`. you can add an `Origin` property to docset if this method is to be shared.